test/e2e/cert_manager_test.go

```diff
@@ -139,3 +144,157 @@ func TestCertManagerProvider(t *testing.T) {
+			TLS: &ecv1alpha1.TLSCertificate{
+				Provider: "cert-manager",
+				ProviderCfg: ecv1alpha1.ProviderConfig{
+					CertManagerCfg: &ecv1alpha1.ProviderCertManagerConfig{
+						CommonConfig: ecv1alpha1.CommonConfig{
+							CommonName:       "etcd-operator-system",
+							ValidityDuration: "90h",
+						},
+						IssuerKind: cmIssuerType,
+						IssuerName: cmIssuerName,
+					},
+				},
+			},
+		},
+	}
+
+	feature.Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+		client := cfg.Client()
+		_ = appsv1.AddToScheme(client.Resources().GetScheme())
+		_ = corev1.AddToScheme(client.Resources().GetScheme())
+		_ = certv1.AddToScheme(client.Resources().GetScheme())
+		_ = apiextensionsV1.AddToScheme(client.Resources().GetScheme())
+
+		var issuer certv1.ClusterIssuer
+		if err := client.Resources().Get(ctx, cmIssuerName, cmCertificateNamespace, &issuer); err != nil {
+			if k8serrors.IsNotFound(err) {
+				t.Logf("No cert-manager selfsigned issuer found, creating clusterIssuer: %s", cmIssuerName)
+				// Create cert-manager issuer
+				if err := client.Resources().Create(ctx, cmIssuer); err != nil {
+					t.Fatalf("unable to create cert-manager selfsigned issuer: %s", err)
+				}
+			} else {
+				t.Fatalf("Failed to get cert-manager selfsigned issuer: %s", cmIssuerName)
+			}
+			t.Logf("Created clusterIssuer: %s", cmIssuerName)
+		}
+
+		// create etcd cluster
+		if err := client.Resources().Create(ctx, etcdCluster); err != nil {
+			t.Fatalf("unable to create etcd cluster: %s", err)
+		}
+
+		// get etcd cluster object
+		var ec ecv1alpha1.EtcdCluster
+		if err := client.Resources().Get(ctx, etcdClusterName, namespace, &ec); err != nil {
+			t.Fatalf("unable to fetch etcd cluster: %s", err)
+		}
+
+		return ctx
+	})
+
+	feature.Assess("Check certificates exist",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			client := c.Client()
+			// checks if client, server, peer certificates are created in the respective namespace
+			if err := wait.For(
+				func(context.Context) (bool, error) {
+					return validateResourceExists(ctx, client, etcdClusterName, namespace, "certificate")
+				},
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(10*time.Second),
+			); err != nil {
+				t.Fatalf("timed out waiting for certificate: %s", err)
+			}
+
+			return ctx
+		},
+	)
+
+	feature.Assess("Check certificate secrets exist",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			client := c.Client()
+			// checks if corresponding client, server, peer secrets are created in the respective namespace
+			if err := wait.For(
+				func(context.Context) (bool, error) {
+					return validateResourceExists(ctx, client, etcdClusterName, namespace, "secret")
+				},
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(10*time.Second),
+			); err != nil {
+				t.Fatalf("timed out waiting for certificate: %s", err)
+			}
+			return ctx
+		},
+	)
+
```

Comment:
Please also verify that we can access the etcdcluster via client certificate. 
- write a key/value pair and read it back

Reply:
Added a check within the `TestClusterCertCreation`